common/tools/dev-tool/src/util/customization/imports.ts

```diff
@@ -1,68 +1,164 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license
 
-import { ImportDeclaration } from "ts-morph";
+import { Identifier, ImportDeclaration, ImportSpecifier, SourceFile, ts } from "ts-morph";
 import { getCustomizationState } from "./state";
 import * as path from "path";
 
 export function augmentImports(
   originalImports: Map<string, ImportDeclaration>,
-  customImports: ImportDeclaration[]
+  customImports: ImportDeclaration[],
+  originalFile: SourceFile
 ) {
   const { customDir, originalDir } = getCustomizationState();
-  const importMap: Map<string, ImportDeclaration> = new Map();
-
-  for (const [, value] of originalImports) {
-    const module = value.getModuleSpecifier().getText();
-    importMap.set(module, value);
+  const importMap = new Map<string, ImportDeclaration>();
+  for (const [moduleSpecifier, importDecl] of originalImports) {
+    importMap.set(moduleSpecifier, importDecl);
   }
 
-  for (const customImport of customImports) {
-    const module = customImport.getModuleSpecifier().getText();
+  const removedModules = removeDuplicateIdentifiers(
+    Array.from(originalImports.values()),
+    customImports
+  );
+  removedModules.forEach(importMap.delete.bind(importMap));
```

Comment:
nitpick: I enjoy one liners myself but this feels a bit overkill and we may want to prioritize readability/simplicity?